ros/src/waypoint_updater/waypoint_updater.py

```diff
@@ -33,23 +34,43 @@ def __init__(self):
 
         rospy.Subscriber('/current_pose', PoseStamped, self.pose_cb)
         rospy.Subscriber('/base_waypoints', Lane, self.waypoints_cb)
+        rospy.Subscriber('/traffic_waypoint', Int32, self.traffic_cb)
+        rospy.Subscriber('/obstacle_waypoint', Lane, self.obstacle_cb)
 
-        # TODO: Add a subscriber for /traffic_waypoint and /obstacle_waypoint
-
-        self.final_waypoints_pub = rospy.Publisher('final_waypoints', Lane,
-                                                   queue_size=1)
+        self.pub = rospy.Publisher('final_waypoints', Lane, queue_size=1)
 
         # TODO: Add other member variables you need below
+        self.world = None
+        self.ego = None
 
+        self.publish()
         rospy.spin()
 
+    def publish(self):
+        rate = rospy.Rate(5)
+        while not rospy.is_shutdown():
+            idx = self.find_next_waypoint()
+            if idx > -1:
+                rospy.loginfo("Next waypoint: {}".format(idx))
+                waypoints = self.world.waypoints + self.world.waypoints
```

Comment:
Can this be done outside the loop?

Reply:
I meant line 55. How can I comment on a single line of code, or in a specific set of lines?

Reply:
Good catch! About the comments, I believe you can only add comments on a single line...

Reply:
Yes, they are read from a static CSV file so they are just published over and over with the same information